File: src/main/java/com/yemao/users/service/ExperienceService.java
package com.yemao.users.service;

import com.yemao.users.repository.ExperienceRepository;
import com.yemao.users.models.Experience;
import com.yemao.utility.Util;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
public class ExperienceService {

    @Autowired
    private ExperienceRepository expRep;

    @Autowired
    private UserService userServ;

    public List<Experience> findByUid(String uid) {
        return expRep.findByUid(uid);
    }

    public void save(Experience exp) {
        exp.setCreate_time(Util.now());
        exp.setId(Util.uuid());
        expRep.save(exp);
    }

    public void saveOrUpdate(Experience userExperience) {
        expRep.saveOrUpdate(userExperience);
    }

    public void delete(Experience userExperience) {
        expRep.delete(userExperience);
    }

    public void update(Experience userExperience) {
        expRep.update(userExperience);
    }

    public List<Experience> sort(List<Experience> T) {
        return expRep.sort(T);
    }

    public List<Experience> findAll() {
        return expRep.findAll();
    }

    public List<Experience> findBy(String[] keys, String[] values) {
        return expRep.findBy(keys, values);
    }

    public List<Experience> findBy(String key, String value) {
        return expRep.findBy(key, value);
    }

    public List<Experience> findBy(String key, String value, boolean isLikeQuery) {
        return expRep.findBy(key, value, isLikeQuery);
    }

    public List<Experience> findBy(String[] keys, String[] values, boolean isLikeQuery) {
        return expRep.findBy(keys, values, isLikeQuery);
    }

    public Experience findById(String id) {
        return expRep.findById(id);
    }

    public Experience findByIds(Map<String, String> idAndValues) {
        return expRep.findByIds(idAndValues);
    }

    public List<Experience> findBy(Map<String, String> attr, boolean likeQuery) {
        return expRep.findBy(attr, likeQuery);
    }





}

File: src/main/kotlin/com/yemao/users/service/experience.service.kt
package com.yemao.users.service

import com.yemao.repository.ExperienceRepository
import com.yemao.users.models.Experience
import com.yemao.utility.Util
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ExperienceService {

    @Autowired lateinit var expRep: ExperienceRepository

    @Autowired lateinit var userServ: UserService

    fun findByUid(uid: String): List<Experience> {
        return expRep.findByUid(uid)
    }

    fun save(exp: Experience) {
        exp.create_time = Util.now()
        exp.id = Util.uuid()
        expRep.save(exp)
    }

    fun saveOrUpdate(experience: Experience) {
        expRep.saveOrUpdate(experience)
    }

    fun delete(experience: Experience) {
        expRep.delete(experience)
    }

    fun update(experience: Experience) {
        expRep.update(experience)
    }

    fun sort(T: List<Experience>): List<Experience> {
        return expRep.sort(T)
    }

    fun findAll(): List<Experience> {
        return expRep.findAll()
    }

    fun findBy(keys: Array<String>, values: Array<String>): List<Experience> {
        return expRep.findBy(keys, values)
    }

    fun findBy(key: String, value: String): List<Experience> {
        return expRep.findBy(key, value)
    }

    fun findBy(key: String, value: String, isLikeQuery: Boolean): List<Experience> {
        return expRep.findBy(key, value, isLikeQuery)
    }

    fun findBy(keys: Array<String>, values: Array<String>, isLikeQuery: Boolean): List<Experience> {
        return expRep.findBy(keys, values, isLikeQuery)
    }

    fun findById(id: String): Experience {
        return expRep.findById(id)
    }

    fun findByIds(idAndValues: Map<String, String>): Experience {
        return expRep.findByIds(idAndValues)
    }

    fun findBy(attr: Map<String, String>, likeQuery: Boolean): List<Experience> {
        return expRep.findBy(attr, likeQuery)
    }


}